### ISSUE TITLE
containerize with conda fails, missing install.sh

### ISSUE THREAD

**Issue by smidm** [bentoml/BentoML#2666]
**Describe the bug**

`bentoml containerize` with conda options in `bentofile.yaml` fails with `chmod: cannot access '/home/bentoml/bento/env/python/install.sh': No such file or directory`.

`bentofile.yaml` with:
```
python:
      packages: 
      - scikit-learn
```
works without issues.

**To Reproduce**

`sample_bentofile.yaml`:
```
service: "sample_service:svc"
include:
    - "*.py"  
conda:
    dependencies:
    - python=3.8.13
    - pip    
    pip:
    - scikit-learn
```

`sample_service.py`:
```
import numpy as np
import bentoml
from bentoml.io import NumpyNdarray

iris_clf_runner = bentoml.sklearn.get("iris_clf:latest").to_runner()

svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])

@svc.api(input=NumpyNdarray(), output=NumpyNdarray())
def classify(input_series: np.ndarray) -> np.ndarray:
    result = iris_clf_runner.predict.run(input_series)
    return result
```

1. `bentoml build -f sample_bentofile.yaml`
2. `bentoml containerize iris_classifier:latest --debug --no-cache`
3. fails with:

```
Building BentoML service "iris_classifier:d4lyn4xzho2n2atv" from build context "/home/ubuntu/experiments/mmdetection/bentoml"
Packing model "iris_clf:7o4gi2hvzshtkatv"

██████╗░███████╗███╗░░██╗████████╗░█████╗░███╗░░░███╗██╗░░░░░
██╔══██╗██╔════╝████╗░██║╚══██╔══╝██╔══██╗████╗░████║██║░░░░░
██████╦╝█████╗░░██╔██╗██║░░░██║░░░██║░░██║██╔████╔██║██║░░░░░
██╔══██╗██╔══╝░░██║╚████║░░░██║░░░██║░░██║██║╚██╔╝██║██║░░░░░
██████╦╝███████╗██║░╚███║░░░██║░░░╚█████╔╝██║░╚═╝░██║███████╗
╚═════╝░╚══════╝╚═╝░░╚══╝░░░╚═╝░░░░╚════╝░╚═╝░░░░░╚═╝╚══════╝

Successfully built Bento(tag="iris_classifier:d4lyn4xzho2n2atv")
ubuntu@ip-172-31-12-44 ~/e/m/bentoml (master)> bentoml containerize iris_classifier:latest --debug --no-cache                                                       (bentotest) 
Building docker image for Bento(tag="iris_classifier:d4lyn4xzho2n2atv")...
[+] Building 14.0s (17/22)                                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                                       0.0s
 => => transferring dockerfile: 2.87kB                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4-labs                                                                                                          0.1s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4-labs@sha256:b50ad4af81d1c76ab7c0e1ffc216909e7adc23e99910243e1c88331c2a8ef52d                                     0.0s
 => [internal] load build definition from Dockerfile                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/continuumio/miniconda3:latest                                                                                                  13.6s
 => CACHED [internal] settings cache mount permissions                                                                                                                     0.0s
 => CACHED [cached 1/1] FROM docker.io/continuumio/miniconda3:latest@sha256:977263e8d1e476972fddab1c75fe050dd3cd17626390e874448bd92721fd659b                               0.0s
 => [internal] load build context                                                                                                                                          0.0s
 => => transferring context: 33.62kB                                                                                                                                       0.0s
 => [stage-1  2/13] RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache                 0.3s
 => [stage-1  3/13] RUN --mount=type=cache,from=cached,sharing=shared,target=/var/cache/apt     --mount=type=cache,from=cached,sharing=shared,target=/var/lib/apt  apt-g  11.3s
 => [stage-1  4/13] RUN rm -rf /var/lib/{apt,cache,log}                                                                                                                    0.4s
 => [stage-1  5/13] RUN groupadd -g 1034 -o bentoml && useradd -m -u 1034 -g 1034 -o -r bentoml                                                                            0.4s 
 => [stage-1  6/13] RUN mkdir /home/bentoml/bento && chown bentoml:bentoml /home/bentoml/bento -R                                                                          0.5s 
 => [stage-1  7/13] WORKDIR /home/bentoml/bento                                                                                                                            0.0s 
 => [stage-1  8/13] COPY --chown=bentoml:bentoml . ./                                                                                                                      0.0s 
 => ERROR [stage-1  9/13] RUN --mount=type=cache,mode=0777,target=/root/.cache/pip     chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento  0.4s 
------
 > [stage-1  9/13] RUN --mount=type=cache,mode=0777,target=/root/.cache/pip     chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento/env/python/install.sh:
#0 0.325 chmod: cannot access '/home/bentoml/bento/env/python/install.sh': No such file or directory
------
error: failed to solve: executor failed running [/bin/sh -c chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento/env/python/install.sh]: exit code: 1
Failed building docker image: Command '['docker', 'buildx', 'build', '--progress', 'auto', '--tag', 'iris_classifier:d4lyn4xzho2n2atv', '--file', 'env/docker/Dockerfile', '--load', '--no-cache', '.']' returned non-zero exit status 1.
```

**Environment:**
 - Ubuntu 18.04.6 LTS
 - Python 3.8.13
 - bentoml, version 1.0.0rc2.post24+g25b6e63

**Comment by aarnphm** [comment]
Can you try `docker builder prune --filter type=exec.cachemount` and `docker volume prune` and try again?

assuming you don't have any crucial volume in docker.

**Comment by smidm** [comment]
I've tried and still the same results.

**Comment by aarnphm** [comment]
Can you try with rc3 releases to see if you still see this issue. I can't reproduce it.

**Comment by MathiasGruber** [comment]
I'm using the rc3 and getting same error:

`python
service: "service:svc"
include:
  - "*.py"
conda:
  environment_yml: "./artifacts/loaded_model/conda.yaml"
`

`[+] Building 651.2s (18/21)                                                                                                                                                                                                               
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 2.98kB                                                                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4-labs                                                                                                                                                                    0.6s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4-labs@sha256:b50ad4af81d1c76ab7c0e1ffc216909e7adc23e99910243e1c88331c2a8ef52d                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => [internal] load metadata for docker.io/continuumio/miniconda3:latest                                                                                                                                                           650.4s
 => [internal] load build context                                                                                                                                                                                                    0.0s
 => => transferring context: 22.86kB                                                                                                                                                                                                 0.0s
 => [internal] settings cache mount permissions                                                                                                                                                                                      0.0s
 => [cached 1/1] FROM docker.io/continuumio/miniconda3:latest@sha256:977263e8d1e476972fddab1c75fe050dd3cd17626390e874448bd92721fd659b                                                                                                6.0s
 => => resolve docker.io/continuumio/miniconda3:latest@sha256:977263e8d1e476972fddab1c75fe050dd3cd17626390e874448bd92721fd659b                                                                                                       0.0s
 => => sha256:977263e8d1e476972fddab1c75fe050dd3cd17626390e874448bd92721fd659b 1.36kB / 1.36kB                                                                                                                                       0.0s
 => => sha256:58b1c7df8d69655ffec017ede784a075e3c2e9feff0fc50ef65300fc75aa45ae 953B / 953B                                                                                                                                           0.0s
 => => sha256:ce7d119281a1f4685ce6ca66b355c88baa44522ac6a54aee86be96d14ab6dfda 4.36kB / 4.36kB                                                                                                                                       0.0s
 => => sha256:1a23c9d790a34c5bb13dbaf42e0ea2a555e089aefed7fdfa980654f773b39b39 50.06MB / 50.06MB                                                                                                                                     2.2s
 => => sha256:22a6fc63b9b529f00082379be512f0ca1c7a491872396994cf59b47e794c5e09 60.29MB / 60.29MB                                                                                                                                     4.0s
 => => extracting sha256:1a23c9d790a34c5bb13dbaf42e0ea2a555e089aefed7fdfa980654f773b39b39                                                                                                                                            1.7s
 => => extracting sha256:22a6fc63b9b529f00082379be512f0ca1c7a491872396994cf59b47e794c5e09                                                                                                                                            1.8s
 => [stage-6  1/11] RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache                                                                           0.4s
 => [stage-6  2/11] RUN --mount=type=cache,from=cached,sharing=shared,target=/var/cache/apt     --mount=type=cache,from=cached,sharing=shared,target=/var/lib/apt  apt-get update -y     && apt-get install -q -y --no-install-rec  49.0s
 => [stage-6  3/11] RUN rm -rf /var/lib/{apt,cache,log}                                                                                                                                                                              0.2s
 => [stage-6  4/11] RUN groupadd -g 1034 -o bentoml && useradd -m -u 1034 -g 1034 -o -r bentoml                                                                                                                                      0.4s
 => [stage-6  5/11] RUN mkdir /home/bentoml/bento && chown bentoml:bentoml /home/bentoml/bento -R                                                                                                                                    0.3s
 => [stage-6  6/11] WORKDIR /home/bentoml/bento                                                                                                                                                                                      0.0s
 => [stage-6  7/11] COPY --chown=bentoml:bentoml . ./                                                                                                                                                                                0.0s
 => [stage-6  8/11] RUN --mount=type=cache,mode=0777,target=/opt/conda/pkgs bash <<EOF                                                                                                                                             591.8s
 => ERROR [stage-6  9/11] RUN --mount=type=cache,mode=0777,target=/root/.cache/pip     chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento/env/python/install.sh                                      0.4s 
------                                                                                                                                                                                                                                    
 > [stage-6  9/11] RUN --mount=type=cache,mode=0777,target=/root/.cache/pip     chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento/env/python/install.sh:                                                 
#0 0.411 chmod: cannot access '/home/bentoml/bento/env/python/install.sh': No such file or directory                                                                                                                                      
------                                                                                                                                                                                                                                    
error: failed to solve: executor failed running [/bin/sh -c chmod +x /home/bentoml/bento/env/python/install.sh &&     bash /home/bentoml/bento/env/python/install.sh]: exit code: 1                                                       
Failed building docker image: Command '['docker', 'buildx', 'build', '--progress', 'auto', '--tag', 'iris_classifier:iq5nbeh3okzncasc', '--platform', 'linux/amd64', '--file', 'env/docker/Dockerfile', '--load', '.']' returned non-zero exit status 1.`

**Comment by aarnphm** [comment]
I can't seem to reproduce, even with a completely new environment. Can you send me your bento with
```bentoml export iris_classifier:<tag> .``` If there aren't anything sensitive in there

**Comment by MathiasGruber** [comment]
Absolutely: 
[iris_classifier-iq5nbeh3okzncasc.bento.zip](https://github.com/bentoml/BentoML/files/9038213/iris_classifier-iq5nbeh3okzncasc.bento.zip)

**Comment by aarnphm** [comment]
You might have to repackage the bento again. I see that from the bento you sent me this is the rc2 format. We currently don't have any backwards compatibility across rc releases as it is a pre release.

If possible I suggest you should try the whole workflow again with rc3 release. This include repackaging and containerize the bento.

**Comment by MathiasGruber** [comment]
Ah, indeed, apologies. Indeed building with rc3 fixed the issue 👍

**Comment by smidm** [comment]
I've upgraded to rc3, run `bentoml build` and `bentoml containerize` ending with the same results. Attaching the bento:
[iris_classifier-ifj52sx3rohtkatv.zip](https://github.com/bentoml/BentoML/files/9039046/iris_classifier-ifj52sx3rohtkatv.zip)
.

**Comment by aarnphm** [comment]
If possible, a .bento format would be helpful.

**Comment by aarnphm** [comment]
This is weird.

**Comment by aarnphm** [comment]
I will try to put a PR to address this. Thanks for the information.

**Comment by smidm** [comment]
It's bento, just renamed to zip as it is one of the allowed extensions in comments.

**Comment by aarnphm** [comment]
This issue will now be tracked at #2681

**Comment by aarnphm** [comment]
This issue has been fixed at #2681. This will be addressed at `1.0` release.

**Comment by NoyHanan** [comment]
Hey all, I still having this issue, using macbook air m2.
bento.yaml:

```
service: "src.service_hermes:svc"
labels:
  owner: test
  stage: dev
include:
- "*.py"  # A pattern for matching which files to include in the bento
python:
  packages:  # Additional pip packages required by the service   
  - "scikit-learn==1.3.2"
  - "pandas==2.1.3"
  - "bentoml>=1.0.0"
  - "langchain==0.0.343"
  - "openai==1.3.5"
  - "pinecone-client==2.2.4"
  - "python-dotenv==1.0.0"
  - "tiktoken==0.5.1"
  - "transformers>=4.35.0"
  - "accelerate>=0.21.0"
  - "bitsandbytes>=0.41.0"
  - "scipy==1.11.4"
  - "openllm[vllm]>=0.4.33"
  - "auto-gptq>=0.5.1"
docker:
  # dockerfile_template: "./Dockerfile.template"
  distro: debian
  # setup_script: "./setup.py"
  python_version: "3.10"
```
